spec/registry.md

```diff
@@ -0,0 +1,156 @@
+    <pattern id="currencyCode" regex="[A-Z]{3}"/>
+
+    <function name="number">
+        <description>
+            Format a number.
+            Match a numerical value against CLDR plural categories or against a number literal.
+        </description>
+
+        <matchSignature locales="en">
+            <input title="The number to match." pattern="anyNumber"/>
+            <option name="type" values="cardinal ordinal"/>
+            <option name="minimumIntegerDigits" pattern="positiveInteger"/>
+            <option name="minimumFractionDigits" pattern="positiveInteger"/>
+            <option name="maximumFractionDigits" pattern="positiveInteger"/>
+            <option name="minimumSignificantDigits" pattern="positiveInteger"/>
+            <option name="maximumSignificantDigits" pattern="positiveInteger"/>
+            <match values="one other"/>
+            <match pattern="anyNumber"/>
+        </matchSignature>
+
+        <formatSignature locales="en">
+            <input title="The number to format" pattern="anyNumber"/>
+            <option name="minimumIntegerDigits" pattern="positiveInteger"/>
+            <option name="minimumFractionDigits" pattern="positiveInteger"/>
+            <option name="maximumFractionDigits" pattern="positiveInteger"/>
+            <option name="minimumSignificantDigits" pattern="positiveInteger"/>
+            <option name="maximumSignificantDigits" pattern="positiveInteger"/>
+            <option name="style" readonly="true" values="decimal currency percent unit" default="decimal"/>
+            <option name="currency" readonly="true" pattern="currencyCode"/>
+        </formatSignature>
+    </function>
+</registry>
+```
+
+A localization engineer can then extend the registry by defining the following `customRegistry.xml` file.
+
+```xml
+<?xml version="1.0" encoding="UTF-8" ?>
+<!DOCTYPE registry SYSTEM "./registry.dtd">
+
+<registry>
+    <function name="noun">
+        <description>Handle the grammar of a noun.</description>
+        <formatSignature locales="en">
+            <input title="Noun id"/>
+            <option name="article" values="definite indefinite"/>
+            <option name="plural" values="one other"/>
+            <option name="case" values="nominative genitive" default="nominative"/>
+        </formatSignature>
+    </function>
+
+    <function name="adjective">
+        <description>Handle the grammar of an adjective.</description>
+        <formatSignature locales="en">
+            <input title="Adjective id"/>
+            <option name="article" values="definite indefinite"/>
+            <option name="plural" values="one other"/>
+            <option name="case" values="nominative genitive" default="nominative"/>
+        </formatSignature>
+        <formatSignature locales="en">
+            <input title="Adjective id"/>
+            <option name="article" values="definite indefinite"/>
+            <option name="accord"/>
```

Comment:
The `accord` option doesn't have a pattern or an enumeration, which surprises me given lines 51-52 (which suggest to me that every option has one or the other).

Reply:
Right, this is still WIP and underspec'ed. I'd like to continue the discussion about validating runtime values outside of this PR.